packages/vsce/src/trees/CICSTree.ts

```diff
@@ -583,35 +555,23 @@ export class CICSTree implements TreeDataProvider<CICSSessionTree> {
 
   async updateSession(session: CICSSessionTree) {
     await ProfileManagement.profilesCacheRefresh();
-    const profileToUpdate = await ProfileManagement.getProfilesCache().loadNamedProfile(session.label?.toString()!, "cics");
-
-    const message = {
-      name: profileToUpdate.name,
-      profile: {
-        name: profileToUpdate.name,
-        ...profileToUpdate.profile,
-      },
-    };
-    await this.updateSessionHelper(session, message);
+    const profileCache = await ProfileManagement.getProfilesCache();
+    const profileToUpdate = profileCache.loadNamedProfile(session.label?.toString()!, "cics");
+    const currentProfile = await profileCache.getProfileFromConfig(profileToUpdate.name);
+    await this.updateSessionHelper(currentProfile);
   }
 
-  async updateSessionHelper(session: CICSSessionTree, messageToUpdate?: imperative.IUpdateProfile) {
-    const column = window.activeTextEditor ? window.activeTextEditor.viewColumn : undefined;
-    const panel: WebviewPanel = window.createWebviewPanel("zowe", `Update CICS Profile`, column || 1, { enableScripts: true });
-    panel.webview.html = addProfileHtml(messageToUpdate);
-    await panel.webview.onDidReceiveMessage(async (message) => {
-      try {
-        panel.dispose();
-        const profile = await ProfileManagement.updateProfile(message);
-        const position = this.loadedProfiles.indexOf(session);
-        await ProfileManagement.profilesCacheRefresh();
-        const updatedProfile = await ProfileManagement.getProfilesCache().loadNamedProfile(profile.profile.name, "cics");
-        await this.removeSession(session, updatedProfile, position);
-      } catch (error) {
-        // @ts-ignore
-        window.showErrorMessage(error);
-      }
+  async updateSessionHelper(profile?: imperative.IProfAttrs) {
```

Comment:
A profile is always provided to this method right? Therefore we don't need an optional parameter here.

```suggestion
  async updateSessionHelper(profile: imperative.IProfAttrs) {
```